lib/src/main/java/graphql/nadel/engine/transform/query/NadelQueryTransformer.kt

```diff
@@ -168,8 +168,14 @@ class NadelQueryTransformer private constructor(
     }
 
     private fun getUnderlyingTypeNames(objectTypeNames: Collection<String>): List<String> {
-        return objectTypeNames.map {
-            executionBlueprint.getUnderlyingTypeName(service, overallTypeName = it)
+        return if (executionContext.hints.sharedTypeRenames(service)) {
+            objectTypeNames.map {
+                executionBlueprint.getUnderlyingTypeName(overallTypeName = it)
```

Comment:
This… doesn't feel right.

Sorry I haven't really been following this work, but renames are technically supposed to be per service.

Tbh I don't really remember the full story, but a long time ago I wanted to implement repeated `@renamed` with a `service` argument.